### PR TITLE
doc: add v2.3 release notes draft

### PR DESCRIPTION
**Pull request by dbkinder**
Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by dbkinder** [comment]
I'll merge this to @wenlingz can add the missing fixed and known issues sections

File: doc/release_notes/release_notes_2.3.rst [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+Shared memory based inter-VM communication (ivshmem) is extended
+  ivshmem now supports interrupts (See :ref:`ivshmem-hld`).
+
+Enhanced vUART support
+  Added console support using legacy vUART (0x3F8-like) and
+  added PCI vUART (up to 8) for VM-to-VM communication.
+
+End-to-end secure boot improvement
+  OVMF can be loaded now as two blobs, one for code and the other for data.
+  The code blob can be verified by the Service VM's ``dm-verify`` as
```
Comment (gvancuts) [inline doc/release_notes/release_notes_2.3.rst:51]: Isn't that `dm-verity` rather?

File: doc/release_notes/release_notes_2.3.rst [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+* :ref:`hld-security`
+* :ref:`hld-trace-log`
+* :ref:`hld-virtio-devices`
+* :ref:`ivshmem-hld`
+* :ref:`l1tf`
+* :ref:`modularity`
+* :ref:`sw_design_guidelines`
+* :ref:`rt_industry_ubuntu_setup`
+* :ref:`introduction`
+* :ref:`release_notes_2.2`
```
Comment (gvancuts) [inline doc/release_notes/release_notes_2.3.rst:93]: It's a little odd to call out that we updated the Release Note for v2.2 as part of our v2.3 release.

File: doc/release_notes/release_notes_2.3.rst [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+* :ref:`using_grub`
+* :ref:`using_hybrid_mode_on_nuc`
+* :ref:`using_partition_mode_on_nuc`
+* :ref:`using_windows_as_uos`
+* :ref:`using_zephyr_as_uos`
+* :ref:`vuart_config`
+* :ref:`how-to-enable-secure-boot-for-windows`
+* :ref:`acrn-dm_parameters`
+
+Because we're dropped deprivileged boot mode support,
```
Comment (gvancuts) [inline doc/release_notes/release_notes_2.3.rst:118]: "... we're dropped..." -> "...we've dropped..."

**Pull request merged**